src/java/org/apache/cassandra/cql3/statements/RevokePermissionsStatement.java

```diff
@@ -41,7 +45,25 @@ public RevokePermissionsStatement(Set<Permission> permissions, IResource resourc
 
     public ResultMessage execute(ClientState state) throws RequestValidationException, RequestExecutionException
     {
-        DatabaseDescriptor.getAuthorizer().revoke(state.getUser(), permissions, resource, grantee);
+        IAuthorizer authorizer = DatabaseDescriptor.getAuthorizer();
+        Set<Permission> revoked = authorizer.revoke(state.getUser(), permissions, resource, grantee);
+
+        // We want to warn the client if all the specified permissions have not been revoked and the client did
+        // not specified ALL in the query.
+        if (!revoked.equals(permissions) && !permissions.equals(Permission.ALL))
+        {
+            // We use a TreeSet to guarantee the order for testing
+            String permissionsStr = new TreeSet<>(permissions).stream()
+                                                              .filter(permission -> !revoked.contains(permission))
+                                                              .map(Permission::name)
+                                                              .collect(Collectors.joining(", "));
```

Comment:
As before, we can use `Stream#sort` instead of the `TreeSet`:
```suggestion
            String permissionsStr = permissions.stream()
                                               .filter(permission -> !revoked.contains(permission))
                                               .sorted(Permission::compareTo) // guarantee the order for testing
                                               .map(Permission::name)
                                               .collect(Collectors.joining(", "));
```